examples/StocksCalculator/main.ipynb

```diff
@@ -0,0 +1,310 @@
+    "        }"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cc407946",
+   "metadata": {},
+   "source": [
+    "Functions for fetching and processing data from Yahoo finance API."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b8c9fbd1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def fetch_yahoo_price(ticker, start_date, end_date):\n",
+    "    start_ts = int(time.mktime(start_date.timetuple()))\n",
+    "    end_ts = int(time.mktime(end_date.timetuple()))\n",
+    "    url = f\"https://query1.finance.yahoo.com/v8/finance/chart/{ticker}\"\n",
+    "    params = {\n",
+    "        \"period1\": start_ts,\n",
+    "        \"period2\": end_ts,\n",
+    "        \"interval\": \"1d\"\n",
+    "    }\n",
+    "\n",
+    "    try:\n",
+    "        resp = requests.get(url, params=params, headers=HEADERS)\n",
+    "        data = resp.json()\n",
+    "\n",
+    "        result = data[\"chart\"][\"result\"][0]\n",
+    "        timestamps = result[\"timestamp\"]\n",
+    "        closes = result[\"indicators\"][\"quote\"][0][\"close\"]\n",
+    "\n",
+    "        dates = [datetime.fromtimestamp(ts).strftime('%Y-%m-%d') for ts in timestamps]\n",
+    "        return pd.Series(closes, index=dates)\n",
+    "    except Exception as e:\n",
+    "        print(f\"Error for {ticker}: {e}\")\n",
+    "        return None\n",
+    "\n",
+    "\n",
+    "def get_price_changes(tickers, date_str) -> dict:\n",
+    "    date_obj = datetime.strptime(date_str, \"%Y-%m-%d\")\n",
+    "    day_before = date_obj - timedelta(days=1)\n",
+    "    day_after = date_obj + timedelta(days=1)\n",
+    "\n",
+    "    changes = {}\n",
+    "    for ticker in tickers:\n",
+    "        series = fetch_yahoo_price(ticker, day_before, day_after)\n",
+    "        print(series)\n",
+    "        if series is not None and date_str in series and day_before.strftime(\"%Y-%m-%d\") in series:\n",
+    "            prev = series[day_before.strftime(\"%Y-%m-%d\")]\n",
+    "            curr = series[date_str]\n",
+    "            if prev and curr:\n",
+    "                change = ((curr - prev) / prev) * 100\n",
+    "                changes[ticker] = change\n",
+    "    return changes\n",
+    "\n",
+    "def plot_with_pandas(changes, date_str) -> bytes:\n",
+    "    df = pd.DataFrame(list(changes.items()), columns=[\"Ticker\", \"Change (%)\"])\n",
+    "    df.sort_values(\"Change (%)\", ascending=False, inplace=True)\n",
+    "\n",
+    "    ax = df.plot.bar(\n",
+    "        x=\"Ticker\",\n",
+    "        y=\"Change (%)\",\n",
+    "        title=f\"Stock Change (%) on {date_str}\",\n",
+    "        xlabel=\"Ticker\",\n",
+    "        ylabel=\"Change (%)\",\n",
+    "        grid=True,\n",
+    "        figsize=(10, 7)\n",
+    "    )\n",
+    "\n",
+    "    buf = io.BytesIO()\n",
+    "    ax.figure.savefig(buf, format=\"jpeg\")\n",
+    "    return buf\n"
+   ]
```

Comment:
please add an #ignore  cell where you play with these functions and render the graph directly in jupyter.